### ISSUE TITLE
Error: Uncaught (in promise)

### ISSUE THREAD

**Issue by pkaufi** [SolidStateGroup/bullet-train-js-client#14]
Hello,

I'm currently evaluating different feature toggle services and stumbled upon an error while implementing bullet-train into our angular app. I also checked out the example in this repo and got the same error.

### Steps to reproduce
1. Open dev console while running the angular app
2. change the `env/ironmentID` in `app.component.ts:12:` https://github.com/SolidStateGroup/bullet-train-js-client/blob/006526e1d2df56be53790967858aa0a2c4a1c899/examples/angular/src/app/app.component.ts#L12
to something none existing (e.g. `'1234'`) to force the backend to respond with a `403`. 
3. reload the app

### Actual behavior
```
core.js:6014 ERROR Error: Uncaught (in promise): (intermediate value)(intermediate value).forEach is not a function
    at resolvePromise (zone-evergreen.js:797)
    at zone-evergreen.js:707
    at index.js:1
    at ZoneDelegate.invoke (zone-evergreen.js:359)
    at Object.onInvoke (core.js:39699)
    at ZoneDelegate.invoke (zone-evergreen.js:358)
    at Zone.run (zone-evergreen.js:124)
    at zone-evergreen.js:855
    at ZoneDelegate.invokeTask (zone-evergreen.js:391)
    at Object.onInvokeTask (core.js:39680)
```
`onError` is called, but the error message we get in `onError: (error) => {}` is actually `{message: "(intermediate value)(intermediate value).forEach is not a function"}`

### Expected behavior
no error message shown in console and `onError` being called with the correct error of the api response `{detail: "Invalid or missing Environment Key"}`.

**Comment by kyle-ssg** [comment]
Hey @pkaufi thanks for reporting this, will be looking at this later today. So if I'm understanding correctly, this only seems to occur when receiving an error?

**Comment by pkaufi** [comment]
exactly, this only happens when receiving an error from the api (`403` in case of an invalid `environmentID`). Happy case works fine :+1:

**Comment by kyle-ssg** [comment]
@pkaufi This is fixed in 0.0.82. The full error message will get passed through to the onError callback, also it's worth noting that init returns a promise which will be rejected if the initial request fails.
